### ISSUE TITLE
Would be nice to have `flutter screenrecord`

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#3496]
Even if it only worked on Android for now.

I looked at writing it briefly this morning, but unlike a screen capture which is once and done, a screen recording is likely long running and waits for a keyboard interrupt to stop?

I suspect other CPTs have implemented something like this.

@devoncarew

**Comment by devoncarew** [comment]
Could also be implemented as a `--record` option on the existing screenshot command?

`dart:io` does expose `stdio`, which I believe will let you read from the console.

**Comment by eseidelGoogle** [comment]
--record could work if that's simpler.  Honestly I'd do whatever is simple.  The main value this provides is an (eventual) x-platform support, and in the short term making something easier than:

```
adb screenrecord /sdcard/demo.mp4
^C
adb pull /sdcard/demo.mp4
```

Since not all devs even have adb in their path to begin with. :)

**Comment by tvolkert** [comment]
If we had https://github.com/flutter/flutter/issues/8245, we could just do this in the engine and expose it to flutter_tools

**Comment by eseidelGoogle** [comment]
Looks like it might be possible to screen record from iOS simulators now too:
https://twitter.com/BlueAquilae/status/1024276804981465088
(previously I thought it was only possible from devices).

**Comment by tvolkert** [comment]
Not quite.  This is about creating an abstract way to obtain a screen recording via the Flutter CLI, whereas that wiki page instructs developers on how to do so specifically on Android or iOS.

**Comment by ganeshsp1** [comment]
This feature would be really nice to have and if the screen recording also captures the audio either the internal( for example a song being played) or external or both, it would be much more better.

**Comment by tapizquent** [comment]
This would be nice if it allowed to record screen within the app, such as in a button tap. To then give user the ability to save, email or text it

**Comment by fredlee12345678** [comment]
flutter screenshot --help
Take a screenshot from a connected device.

Usage: flutter screenshot [arguments]
-h, --help                      Print this usage information.
-o, --out=<path/to/file>        Location to write the screenshot.
    --observatory-uri=<URI>     The observatory URI to connect to.
                                This is required when --type is "skia" or "rasterizer".
                                To find the observatory URI, use "flutter run" and look for"An Observatory ... is available at" in the output.

    --type=<type>               The type of screenshot to retrieve.

          [device] (default)    Delegate to the device's native screenshot capabilities. This screenshots the entire screen currently being displayed
                                (including content not rendered by Flutter, like the device status bar).
          [rasterizer]          Render the Flutter app using the rasterizer. Requires --observatory-uri
          [skia]                Render the Flutter app as a Skia picture. Requires --observatory-uri



Why I have no --record option?
My Flutter Version is :
Flutter 1.9.1+hotfix.7

**Comment by bkonyi** [comment]
This seems like it could belong in a package, but I don't think the tool team will have resources to implement a feature like this.
